docs/root/configuration/http_filters/ip_tagging_filter.rst

```diff
@@ -0,0 +1,44 @@
+.. _config_http_filters_ip_tagging:
+
+IP Tagging
+==========
+
+The HTTP IP Tagging filter sets the header *x-envoy-ip-tags* with the string tags for the trusted address from
+:ref:`x-forwarded-for <config_http_conn_man_headers_x-forwarded-for>`:. If there are no tags for an address,
+the header is not set.
+
+Internally, a Level-Compressed trie stores tags for IP Address and CIDR ranges. The algorithm for the Level-Compressed
```

Comment:
For user-facing documentation, perhaps the most important thing to say about the IP Tagging implementation is that it provides a scalable way to compare client IP addresses against large lists of CIDR prefixes efficiently.